### PR TITLE
No more disposing `Image<Rgb24>` input.

### PR DESCRIPTION
**Pull request by salarcode**
There was an issue when the input to `CalcDiff` was already an image of `Image<Rgb24>` in that case the method was disposing the image preventing the reuse of it which is very important for performance.

I've created a fix for that with additional changes:
- No more disposing Image<Rgb24> input.
- New overloads to accept `Image<Rgb24>` as input that can be used for cached images.
- Expose `ConvertRgba32ToRgb24`
- Performance improvements

### CODE REVIEW

**Comment by salarcode** [comment]
I have also added some performance improvement in accessing to pixels that should result in up to 40% performance improvement.

File: ImageSharpCompare/ImageSharpCompare.cs [hunk 1]
```diff
@@ -82,20 +82,45 @@ public static bool ImagesAreEqual(Stream actual, Stream expected)
...
+            {
+                actualPixelaccessableImage = ToRgb24Image(actual, out ownsActual);
+                expectedPixelaccessableImage = ToRgb24Image(expected, out ownsExpected);
+
+                return ImagesAreEqual(actualPixelaccessableImage, expectedPixelaccessableImage);
+            }
+            finally
+            {
+                if (ownsActual)
+                    actualPixelaccessableImage?.Dispose();
```
Comment (stesee) [inline ImageSharpCompare/ImageSharpCompare.cs:100]: Do you have time to add tests that proof the expected behavior?

File: ImageSharpCompare/ImageSharpCompare.cs [hunk 2]
```diff
@@ -82,20 +82,45 @@ public static bool ImagesAreEqual(Stream actual, Stream expected)
         /// <param name="expected"></param>
         /// <returns>True if every pixel of actual is equal to expected</returns>
         public static bool ImagesAreEqual(Image actual, Image expected)
+        {
```
Comment (stesee) [inline ImageSharpCompare/ImageSharpCompare.cs:85]: Wouldnt overloads also solve the issue, where every Image<Rgb24> wont be disposed while Image could still be diposed using "using"s?

**Review by stesee — CHANGES_REQUESTED** [review]
nice work

File: ImageSharpCompare/ImageSharpCompare.cs (hunk 2 above)
Comment (stesee) [reply ImageSharpCompare/ImageSharpCompare.cs:85]: That would be more readable and wont trigger any CA2000 compiler warning.

File: ImageSharpCompare/ImageSharpCompare.cs (hunk 1 above)
Comment (salarcode) [reply ImageSharpCompare/ImageSharpCompare.cs:101]: Not at the moment, I will see if I can add later.

File: ImageSharpCompare/ImageSharpCompare.cs (hunk 2 above)
Comment (salarcode) [reply ImageSharpCompare/ImageSharpCompare.cs:85]: CA should be happy now

**Pull request merged**